### ISSUE TITLE
SPU2-X: Higurashi no Naku Koro ni Matsuri Kakera Asobi (SPLM-66913) voices

### ISSUE THREAD

**Issue by Setlean** [PCSX2/pcsx2#918]
3 years later still broken.

Repost from https://code.google.com/p/pcsx2/issues/detail?id=1365 
**1) Did the game ever work correctly (i.e. not have this problem) on the
Official PCSX2 build or an earlier version of PCSX2 playground?
(If so, please specify the latest pcsx2-playground or Official revision
that last worked.)**
r5418 and up have problems with voice playback. r4600 up to r5417 seem to work.

**2) What steps will reproduce the problem?**
1. Start the game and listen to any spoken lines

**3) What exactly happens when you experience this issue (listing any console
errors or screen output you receive)?**
At random, parts of the spoken lines will not play. Some lines work correctly some of the time, usually only the start or the end (if anything) of a line will play.

**4) What version of PCSX2 are you using? On what operating system? Plugins?**
Normally r5452 on Windows 7, with SPU2-X version r5446. I have tried most versions of the SPU2-X plugin between r4600 and r5460 and anything above r5417 does not work correctly.

**Comment by ssakash** [comment]
This seems to be the commit which cause the issue : https://code.google.com/p/pcsx2/source/detail?r=5418

it's one of @sudonim1 code, maybe he has an idea ?

**Comment by Setlean** [comment]
I don't really know what i'm doing, but I manage to fix it.
Looks like it really doesn't 4 cycles delay.

I changed:
Plugins/SPU2-X/Source Files/SPU2/spu2sys.cpp
line 309:
from **"if (Cycles - PlayCycle < 4)"** to  **"if (Cycles - PlayCycle < 1)"**
line 320:
from **"if((Cycles-PlayCycle)>=4)"** to **"if((Cycles-PlayCycle)>=1)"**

I played for 30-40 minutes and everything seems to be working how it should.

**Comment by ssakash** [comment]
that delay was specifically meant for  DQ5 adventure log music and Le Mans 24 Hours , if you remove the delay then the other games will get undesired behavior. the best solution would be to satisfy both of these games.

**Comment by Setlean** [comment]
Maybe just make a checkbox in SPU2 settings to manually set cycles to 1?
Looks like this bug is Higurashi specific, so it shouldn't be a problem.
Yes, it's not pretty, but I'm just a noob and really have no idea how everything is working here.
Also, from my understanding 4cycles-delay was also a hack in first place.

**Comment by ssakash** [comment]
an optional setting for delay might be possible to circumvent around this and sounds good for debugging around various games, it might be ideal to have something like that temporarily and finally experiment all games and figure out a more general approach. what do you think @refractionpcsx2  ? 

I'll make a patch up for this if you agree :)

**Comment by refractionpcsx2** [comment]
I can't help but feel this delay is patching the symptoms and not fixing the cause. It all looks very hacky.

There's probably a reg setting the games it was intended for are expecting to see something set, but because we are dealing with it so quickly it got unset too quick.

**Comment by Setlean** [comment]
You want to remove cycle delays in the first place and try to find real source of the problem?
It doesn't sound too easy.
We need to be able to set cycle manually and see what is set on 4 cycles and unset on 1 cycle in the moment of bug.
Maybe this way we will able to find what is causing it?

**Comment by refractionpcsx2** [comment]
I don't personally know the flow of SPU2 overly well (i know a little bit) but generally there are statuses connected to the voices, i suspect we are powering through the setting of the buffer for the voice and unsetting the "voice on" flag as it's already been sent to the sound card, but games might be going

> ok play this
> reading your voice flag
> why the f#@k aren't you playing that sound?
> lets try again
> play this
> reading your voice flag
> are you taking the piss?

where at the moment, with the delay its seeing it, so it goes

> ok play this
> reading your voice flag
> ah there you go, playing away, we're all happy now :)

With lemans it might be the other way around, it might be sending a 0 length clip to the voice to make it start and stop, like a purge of the voice, but when it sees 4 cycles in it's playing something, that could be freaking it out, thinking the voice control code is broken and freezing up.

**Comment by Setlean** [comment]
It's opposite with higurashi.
From testing, looks like voiced sentences are split in parts there and with cycles delay it just skip beginning part and start playing from last parts. Sometimes you got to listen full line, but most of the time you just hear ending of the last word in sentence. 
I also found that you can bug out playback if you press repeat before it finish playing last part and hear the full line, but it not always working.
So for now, i think, checkbox for setting cycle-delay in SPU2 settings will be enough.
This way you at least could be able to play the game on most recent releases (it was broken for literally 3 years).

**Comment by ssakash** [comment]
@refractionpcsx2 

would a ini only option instead of on the GUI be better ? @Setlean does have a point though, it'd be nice to observe how these delays influence other games for quicker debugger on games which have quirks on these stuffs.

**Comment by refractionpcsx2** [comment]
Yeah we could do an ini only option,  that's a solution for now :)

**Comment by willkuer** [comment]
The question is if 1 fixes one game and 4 fixes another... What happens at delays of 2,3,5,8 or whatever?

**Comment by refractionpcsx2** [comment]
Well I guess we will find out with the ini option :)

**Comment by ssakash** [comment]
DelayCycles  to 1 also fixes some sound issues on Armored Core : http://forums.pcsx2.net/Thread-Bug-Report-Armored-Core-2-NTSC-U?page=2

time to find a general fix for this issue !
